python/tests/rtl/test_serialize.py

```diff
@@ -230,7 +236,263 @@ def __init__(self, *, name2: str):
 }
 
 
-def test_deserialize_single():
+@pytest.fixture
+def bm():
+    return Model(
+        enum1=Enum1.entry1,
+        model_no_refs1=ModelNoRefs1(name1="model_no_refs1_name"),
+        enum2=Enum2.entry2,
+        model_no_refs2=ModelNoRefs2(name2="model_no_refs2_name"),
+        list_enum1=[Enum1.entry1],
+        list_model_no_refs1=[ModelNoRefs1(name1="model_no_refs1_name")],
+        opt_enum1=Enum1.entry1,
+        opt_model_no_refs1=None,
+        id=1,
+        name="my-name",
+        class_="model-name",
+        finally_=[1, 2, 3],
+    )
+
+
+def test_dict_getitem_simple(bm):
+    assert bm["id"] == bm.id
+    assert bm["name"] == bm.name
+    assert bm["class"] == bm.class_
+    assert bm["finally"] == bm.finally_
+    with pytest.raises(KeyError):
+        bm["class_"]
+    with pytest.raises(KeyError):
+        bm["finally_"]
+
+
+def test_dict_getitem_child(bm):
+    assert bm["model_no_refs1"] == ModelNoRefs1(name1="model_no_refs1_name")
+    assert bm["model_no_refs1"] is bm.model_no_refs1
+    assert bm["model_no_refs1"]["name1"] == "model_no_refs1_name"
+    assert bm["list_model_no_refs1"][0] == ModelNoRefs1(name1="model_no_refs1_name")
+    assert bm["list_model_no_refs1"][0] is bm.list_model_no_refs1[0]
+    assert bm["list_model_no_refs1"][0]["name1"] == "model_no_refs1_name"
+    with pytest.raises(KeyError):
+        bm["opt_model_no_refs1"]
+    with pytest.raises(KeyError):
+        bm["no_such_prop"]
+
+
+def test_dict_getitem_enum(bm):
+    assert bm["enum1"] == "entry1"
+
+
+def test_dict_setitem_simple(bm):
+    bm["id"] = 2
+    assert bm["id"] == 2
+    assert bm.id == 2
+    bm["name"] = "some-name"
+    assert bm["name"] == "some-name"
+    assert bm.name == "some-name"
+    bm["class"] = "some-class"
+    assert bm["class"] == "some-class"
+    assert bm.class_ == "some-class"
+    bm["finally"] = [4, 5, 6]
+    assert bm["finally"] == [4, 5, 6]
+    assert bm["finally"] is bm.finally_
+    with pytest.raises(AttributeError) as exc:
+        bm["foobar"] = 5
+    assert str(exc.value) == "'Model' object has no attribute 'foobar'"
+
+
+def test_dict_setitem_child(bm):
+    bm["model_no_refs1"]["name1"] = "model_no_refs1_another_name"
+    assert bm["model_no_refs1"]["name1"] == "model_no_refs1_another_name"
+
+    # creating new children from dictionaries instantiates a new child model
+    # object from that dict but does not maintain a reference.
+    child_dict = {"name1": "I used a dictionary"}
```

Comment:
Did you discover any cool new words in the dictionary?

Reply:
Last night I looked up the word "especial" on my kindle - didn't realize that was a word in English (just thought it was Spanish)